synapse/lib/node.py

```diff
@@ -46,15 +46,29 @@ def __init__(self, snap, buid=None, rawprops=None):
     def __repr__(self):
         return f'Node{{{self.pack()}}}'
 
-    async def storm(self, text, opts=None, user=None):
+    async def storm(self, text, opts=None, user=None, path=None):
+        '''
+        Args:
+            path (Path):
+                If set, then vars from path are copied into the new runtime, and vars are copied back out into path
+                at the end
+
+        Note:
+            If opts is not None and opts['vars'] is set and path is not None, then values of path vars take precedent
+        '''
         query = self.snap.core.getStormQuery(text)
-        with self.snap.getStormRuntime(opts=opts, user=user) as runt:
+        newopts = dict(opts) if opts else {}
```

Comment:
This will construct a new dict if an empty one was given.  Should be  "if opts is not None"

Reply:
it will return a static dict as opposed to dict({}). Both make a new dict.